lib/Service/AvatarService.php

```diff
@@ -0,0 +1,235 @@
+		}
+		return $avatarFolder;
+	}
+
+	public function getAvatar(Room $room, ?IUser $user, bool $darkTheme = false): ISimpleFile {
+		$token = $room->getToken();
+		$avatar = $room->getAvatar();
+		if ($avatar) {
+			try {
+				$folder = $this->appData->getFolder('room-avatar');
+				if ($folder->fileExists($token)) {
+					$file = $folder->getFolder($token)->getFile($avatar);
+				}
+			} catch (NotFoundException $e) {
+			}
+		}
+		// Fallback
+		if (!isset($file)) {
+			if ($room->getType() === Room::TYPE_ONE_TO_ONE) {
+				$users = json_decode($room->getName(), true);
+				foreach ($users as $participantId) {
+					if ($user instanceof IUser && $participantId !== $user->getUID()) {
+						$avatar = $this->avatarManager->getAvatar($participantId);
+						$file = $avatar->getFile(512, $darkTheme);
+					}
+				}
+			} elseif ($room->getObjectType() === 'file') {
+				$file = new InMemoryFile($token, file_get_contents(__DIR__ . '/../../img/icon-text-white.svg'));
+			} elseif ($room->getObjectType() === 'share:password') {
+				$file = new InMemoryFile($token, file_get_contents(__DIR__ . '/../../img/icon-password-white.svg'));
+			} elseif ($room->getObjectType() === 'emails') {
+				$file = new InMemoryFile($token, file_get_contents(__DIR__ . '/../../img/icon-mail-white.svg'));
+			} elseif ($room->getType() === Room::TYPE_PUBLIC) {
+				$file = new InMemoryFile($token, file_get_contents(__DIR__ . '/../../img/icon-public-white.svg'));
+			} else {
+				$file = new InMemoryFile($token, file_get_contents(__DIR__ . '/../../img/icon-contacts-white.svg'));
+			}
+		}
+		return $file;
+	}
+
+	private function deleteAvatarFromFilesystem(Room $room): void {
+		$token = $room->getToken();
+		$avatarFolder = $this->getAvatarFolder($token);
+		if (count($avatarFolder->getDirectoryListing()) >= 1) {
+			$avatarFolder->delete();
+		} elseif ($room->getAvatar() && $avatarFolder->fileExists($room->getAvatar())) {
+			$avatarFolder->getFile($room->getAvatar())->delete();
+		}
+	}
+
+	public function deleteAvatar(Room $room): void {
+		try {
+			$this->deleteAvatarFromFilesystem($room);
+			$this->roomService->setAvatar($room, null);
+		} catch (NotFoundException $e) {
+		}
+	}
+
+	public function roomHasAvatar(Room $room): bool {
+		try {
+			$folder = $this->appData->getFolder('room-avatar');
+			if ($folder->fileExists($room->getToken())) {
+				return true;
+			}
+		} catch (NotFoundException $e) {
+		}
+		return $room->getType() === Room::TYPE_ONE_TO_ONE;
+	}
+
+	public function getAvatarVersion(Room $room, ?string $userId): string {
+		if ($room->getType() === Room::TYPE_ONE_TO_ONE && $userId) {
+			return (string) $this->config->getUserValue($userId, 'avatar', 'version', '0');
+		}
+		return (string) ($this->cache->get($room->getToken() . '.avatarVersion') ?? 0);
+	}
+
+	public function getAvatarUrl(Room $room): string {
+		$arguments = [
+			'token' => $room->getToken(),
+			'apiVersion' => 'v1',
+		];
+		if ($avatar = $room->getAvatar()) {
```

Comment:
```suggestion
		$avatar = $room->getAvatar()
		if ($avatar) {
```

We don't do inline assignments in Nextcloud